### ISSUE TITLE
Bug: normalizer sometimes makes volume too low

### ISSUE THREAD

**Issue by mpogue2** [mpogue2/SquareDesk#1096]
I think it's because many songs have a single high-volume glitch, and since the normalizer is peak-based, the whole song gets normalized too low.

Probably need something better than "peak", that can also be calculated quickly.  
Maybe a few samples of RMS, and see whether they all agree?

Example: Rockin' Pneumonia and Boogie Woogie flu volume is too low.

**Comment by mpogue2** [comment]
Rockin' Pneumonia is normalized to a slightly lower volume, when indeed most of the track already has a low volume, way lower than other songs.  RMS = -21dB, whereas What Could Be Better Than This is about RMS = -12.4dB, about 8dB higher volume!

**Comment by mpogue2** [comment]
edited: Pneumonia.mp3 has the final drum attenuated, so no huge peak at the end.
audiodecoder L1583 shows the wholeSongPeak, which is 0.65 for Pneumonia.
So, we should get a bump of 1/0.65.
But, What Could Be Better Than This is 0.89, and we seem to get a bigger bump with that than with Pneumonia.

This seems wrong to me.

**Comment by mpogue2** [comment]
OK, both songs were actually played and plotted correctly.

Rockin' Pneumonia has something special going on -- the song is mostly around 0.4, but it has a high-amplitude (1.0) drum hit near the end.    So, the max is near 1.0, and almost no upward normalization is gonna happen.

We really need a compressor here rather than a simple "max normalizes to 1.0", but I am reluctant to implement such a thing, since it would need tuning on every song for attack/decay/pre-gain/makeup-gain, etc.

So, unfortunately, the best solution in the short term is just to run a real compressor on Rockin' Pneumonia OFFLINE.

**Comment by mpogue2** [comment]
OK, I used a couple of VST's on Rockin' Pneumonia, and it worked GREAT.

Here's the effects chain I used:
- TDR Nova Parallel Dynamic Equalizer (pushed up the bass by 3dB, highs by 6dB)
- LoudMax (Thomas Mundt, https://loudmax.blogspot.com) = sounds GREAT, no artifacts!  Threshold -10.4dB, Out: -1.6dB
- Youlean Loudness Meter (https://youlean.co/youlean-loudness-meter/)

Problem solved for this song.  The normalizer in SquareDesk does OK on most songs, but it is not a 100% substitute for these VST's.
